test/routes.js

```diff
@@ -2187,3 +2187,100 @@ test('if ide has plugin set, serve config.js with the correct endpoint', async (
   t.equal(res.body.toString(), 'window.GRAPHQL_ENDPOINT = \'/something/app/graphql\';\n' +
     'window.GRAPHIQL_PLUGIN_LIST = []')
 })
+
+test('GET graphql endpoint with invalid additionalRouteProp constraint', async (t) => {
+  const app = Fastify()
+  const schema = `
+    type Query {
+      add(x: Int, y: Int): Int
+    }
+  `
+  app.register(GQL, {
+    schema,
+    additionalRouteProps: {
+      constraints: {
+        host: 'auth.fastify.dev'
+      }
+    }
+  })
+
+  const res = await app.inject({
+    method: 'GET',
+    url: '/graphiql',
+    headers: {
+      Host: 'fail.fastify.dev'
+    }
+  })
+
+  t.equal(res.statusCode, 404)
+})
+
+test('GET graphql endpoint with valid additionalRouteProp constraint', async (t) => {
+  const app = Fastify()
+  const schema = `
+    type Query {
+      add(x: Int, y: Int): Int
+    }
+  `
+  app.register(GQL, {
+    schema,
+    additionalRouteProps: {
+      constraints: {
+        host: 'auth.fastify.dev'
+      }
+    }
+  })
+
+  const query = '{ add(x: 2, y: 2) }'
+
+  const res = await app.inject({
+    method: 'GET',
+    url: '/graphql',
+    headers: {
+      Host: 'auth.fastify.dev',
+      'Content-Type': 'text/plain'
+    },
+    query: {
+      query
+    }
+  })
+
+  t.equal(res.statusCode, 200)
+})
+
+test('GET graphql endpoint, additionalRouteProps should ignore custom handler', async (t) => {
+  const app = Fastify()
+  const schema = `
+    type Query {
+      add(x: Int, y: Int): Int
+    }
+  `
+
+  let executed = false
```

Comment:
for this type of async tests, using t.plan is preferred. in my opinion this is also the only case in which you want to use t.plan, but in such cases, I prefer to use it versus setting a flag like you did

Reply:
wouldn't a t.plan wait for the handler to execute? 

i don't want it to execute here , hence the flag

Reply:
t.plan would cause the test runner to wait to finish the test until all planned assertions have run, which is what you're trying to accomplish here

Reply:
ok fine, as clarified, you're testing that the handler has not executed. this solution works then